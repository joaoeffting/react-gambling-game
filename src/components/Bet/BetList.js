import React from 'react';
import styled from 'styled-components';
import { Slider } from '../Slider/Slider';
import { BetService } from '../../services/BetService';

import footBallIcon from '../../images/icons/football.png';
import tennisIcon from '../../images/icons/tennis.png';
import basketBallIcon from '../../images/icons/basketball.png';
import deFaultIcon from '../../images/icons/default.png';

const ICONS = {
    FOOTBALL: footBallIcon,
    TENNIS: tennisIcon,
    BASKETBALL: basketBallIcon
}

const BetContainer = styled.div`
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    background-color: #222222;
    padding: 20px 0;
    height: 190px;
`;

const Points = styled.div`
    color: #FFCB00;
    font-size: 22px;
`;

const Infos = styled.div`
    display: flex;
    align-items: center;
    margin: 20px 0;
    color: #FFF;
    font-size: 16px;
`;

const SportImage = styled.img`
    margin-right: 10px;
`;

const GameTime = styled.div`
    margin-bottom: 20px;
    color: #868170;
`;

const BetLink = styled.a`
    background-color: #00B438;
    width: 100px;
    height: 30px;
    display: flex;
    align-items: center;
    justify-content: center;
    text-decoration: none;
    color: #FFF;
    border-radius: 5px;
    font-weight: bold;
`;

export const BetList = ({ liveEvents }) => {
    
    if (liveEvents.length === 0) 
        return (<div> Loading Bets ... </div>)

    const events = liveEvents.map(event => {
        return (
            <BetContainer key={event.event.name}>
                {
                    event.liveData.score && 
                    <Points>
                        {event.liveData.score.home} - {event.liveData.score.away}
                    </Points>
                }
                <Infos>
                    <SportImage src={ICONS[event.event.sport] || deFaultIcon} alt="sport type icon" />
                    {event.event.homeName} - {event.event.awayName}
                </Infos>
                <GameTime>
                    {BetService.getBetDateFormatted(event.event.start)}
                </GameTime>
                <BetLink 
                    href={`https://www.unibet.com/betting#/event/live/${event.event.id}`}
                    target="_blank"
                >
                    Place a bet
                </BetLink>
            </BetContainer>
        )
    });

    return (<Slider>{events}</Slider>);
}