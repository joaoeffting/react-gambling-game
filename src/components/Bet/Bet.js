import React, { Component } from 'react';
import styled from 'styled-components';
import { BetService } from '../../services/BetService';
import { BetList } from './BetList';

const BetContent = styled.div`
	width: 900px;
	text-align: left;
	margin: 16px auto;
`;

const Article = styled.article`
    padding: 8px 16px 16px;
    background: #fff;
    border-radius: 4px;
    overflow: hidden;
`;

const H1 = styled.h1`
    font-size: 22px;
`;

const Preamble = styled.p`
    font-weight: bold;
`;

const Aside = styled.aside`
    float: right;
    width: 220px;
    border-left: solid 1px #ccc;
    margin-left: 8px;
    padding: 0 0 8px 16px;
`;

const H2 = styled.h2`
    font-size: 15px;
    margin: 4px 0;
`;

export class Bet extends Component {
    constructor(props) {
        super(props);
        this.state = {
            liveEvents: []
        }
    }
    componentDidMount() {
        this.loadBets();
    }
    
    loadBets = async () => {
        try {
            let liveEvents = await BetService.loadBets();
            this.setState({ liveEvents });
        } catch (e) {
            console.log(e);
        }
    }
    render() {
        return (
            <BetContent>
                <Article>
                    <H1>Live matches</H1>
                    <Preamble>
                        Here is a list of matches that are live right now.
                    </Preamble>
                    <Aside>
                        <H2>Live betting</H2>
                        <p>Place your bets as the action unfolds. We offer a wide selection of live betting events and you can place both single and combination bets.</p>
                        <p>You will be able to see an in-play scoreboard with the current result and match stats, while on selected events you will also be able to watch the action live with Unibet TV on the desktop site.</p>
                    </Aside>
                    <BetList liveEvents={this.state.liveEvents} />
                </Article>
            </BetContent>
        )
    }
}