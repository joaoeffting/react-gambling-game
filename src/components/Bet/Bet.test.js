import React from 'react';
import { Bet } from './Bet';
import renderer from 'react-test-renderer';

test('Bet match snapshot', () => {
  const component = renderer.create(
    <Bet />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});