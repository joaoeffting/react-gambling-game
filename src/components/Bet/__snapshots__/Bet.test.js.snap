// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bet match snapshot 1`] = `
<div
  className="sc-bZQynM hOnkKl"
>
  <article
    className="sc-gzVnrw kBMDxl"
  >
    <h1
      className="sc-htoDjs bSLqlL"
    >
      Live matches
    </h1>
    <p
      className="sc-dnqmqq ylglt"
    >
      Here is a list of matches that are live right now.
    </p>
    <aside
      className="sc-iwsKbI hFdZDs"
    >
      <h2
        className="sc-gZMcBi hHfMSJ"
      >
        Live betting
      </h2>
      <p>
        Place your bets as the action unfolds. We offer a wide selection of live betting events and you can place both single and combination bets.
      </p>
      <p>
        You will be able to see an in-play scoreboard with the current result and match stats, while on selected events you will also be able to watch the action live with Unibet TV on the desktop site.
      </p>
    </aside>
    <div>
       Loading Bets ... 
    </div>
  </article>
</div>
`;
